### ISSUE TITLE
Excellent work, but it doesn't perform well in certain cases.

### ISSUE THREAD

**Issue by Zhzet9611** [little-misfit/GRAG-Image-Editing#7]
Excellent work, but it doesn't perform well in certain cases. Could it be that the prompts are problematic?

model: qwen image edit 2509_4bit
lora: qwen image edit 4steps lightning lor
Case1(pass)
prompt: change color from red to blue
![Image](https://github.com/user-attachments/assets/269075ec-64c7-4903-96e8-0e96a939771a)
![Image](https://github.com/user-attachments/assets/11a808a1-22c7-4b4b-9f97-393d6fa3b861)
![Image](https://github.com/user-attachments/assets/048626de-3a36-4900-b6eb-af818d50820b)
![Image](https://github.com/user-attachments/assets/4a591801-277c-4045-b659-348ea03c807c)
![Image](https://github.com/user-attachments/assets/00fdc886-7768-4942-9244-59a33e6e9e98)
![Image](https://github.com/user-attachments/assets/31b03fb8-4814-4b1d-a29b-96120bb9327f)

case2(faild)
Change the animal from  tiger to a cat

![Image](https://github.com/user-attachments/assets/2d9cb096-3ece-4352-8aaf-bbbf3472aff6)
![Image](https://github.com/user-attachments/assets/415581aa-b2f3-4b58-a528-4e195ae141e6)
![Image](https://github.com/user-attachments/assets/ab15bc93-0ef8-414e-8115-bc4b1ad6fb52)
![Image](https://github.com/user-attachments/assets/48256b8d-7274-4d8a-819d-129ec6c6f52a)
![Image](https://github.com/user-attachments/assets/d778e65d-a6ba-45c5-afec-c844fff394ee)

it cant change the tiger to a cat ...

wo GRAG 

<img width="1024" height="1024" alt="Image" src="https://github.com/user-attachments/assets/986b3163-c1fb-4126-aba2-6d76ff0bca17" />

**Comment by little-misfit** [comment]
Thank you for your support. 🧐Regarding the issue you raised, I think there may be two reasons: (a) your use of lighting-lora may have changed the distribution of features, making our method unsuitable; (b) our experiments have shown that for targeted modification tasks, you can appropriately adjust (increase) the lambda parameter to achieve the desired effect. And you can also provide a link to the code and lora you used, and I will respond further after reproducing the problem.

**Comment by Zhzet9611** [comment]
Thank you for your reply！ Below is my code and LoRa link.
code:
import os
from PIL import Image
import torch
import sys
import json
import tqdm
from pathlib import Path
from termcolor import colored
from diffusers import QwenImageEditPipeline
from hacked_models.scheduler import FlowMatchEulerDiscreteScheduler
from hacked_models.pipeline import QwenImageEditPipeline
from hacked_models.models import QwenImageTransformer2DModel
import sys
import json
import tqdm
import argparse
from hacked_models.utils import *

seed_everything(42)

parser = argparse.ArgumentParser() 

parser.add_argument("--model_path", type=str, default="Qwen/Qwen-Image-Edit")
parser.add_argument("--image_path", type=str, required=True)
parser.add_argument("--edit_prompt", type=str, required=True)
parser.add_argument("--out_path", type=str, default='./results')
parser.add_argument("--cond_b", type=float, required=True)
parser.add_argument("--cond_delta", type=float, required=True)



args = parser.parse_args()

scheduler = FlowMatchEulerDiscreteScheduler.from_pretrained(
    os.path.join(args.model_path, "scheduler"),
    torch_dtype=torch.bfloat16,
)


transformer = QwenImageTransformer2DModel.from_pretrained(
    os.path.join(args.model_path, "transformer"),
    torch_dtype=torch.bfloat16,
)

pipeline = QwenImageEditPipeline.from_pretrained(args.model_path, torch_dtype=torch.bfloat16,
                                            scheduler = scheduler,
                                            transformer=transformer,
                                            )
print("pipeline loaded")
pipeline.to("cuda")
pipeline.load_lora_weights(
    "/data/lab/Models/lora/qwen_image",
    weight_name="Qwen-Image-Edit-2509-Lightning-4steps-V1.0-bf16.safetensors",
    adapter_name="lightning"
)
pipeline.set_progress_bar_config(disable=None)


out_path = args.out_path
os.makedirs(out_path,exist_ok=True)

print(colored(out_path,color = "green"))


editing_instruction = args.edit_prompt
input_image = Image.open(args.image_path).convert('RGB').resize((1024,1024))

os.makedirs(os.path.join(out_path),exist_ok=True)

prompt = editing_instruction
inputs = {
    "image": input_image,
    "prompt": prompt,
    "generator": torch.manual_seed(42),
    # "true_cfg_scale": 4.0,
    "true_cfg_scale": 1.0,

    "negative_prompt": " ",
    # "num_inference_steps": 24,
    "num_inference_steps": 4,

    "return_dict": False,
    "grag_scale":[((512,1.0,1.0),(4096,args.cond_b,args.cond_delta))]*60,
}

with torch.inference_mode():
    output = pipeline(**inputs)
    image,x0_images,saved_outputs = output

image[0].save(os.path.join(out_path,f"{args.image_path.split('/')[-1]}_cond_b-{args.cond_b}_cond_delta-{args.cond_delta}.jpg"))

lora link :
[Qwen-Image-Lightning](https://huggingface.co/lightx2v/Qwen-Image-Lightning)
 
Qwen-Image-Edit-2509/Qwen-Image-Edit-2509-Lightning-4steps-V1.0-bf16.safetensors)
